### DISCUSSION TITLE
Server-side vs client-side authentication with Next-auth (caching)

### DISCUSSION THREAD

**Discussion by codinginflow** [nextauthjs/next-auth#10435] [Help]
I need some clarification here.

Let's say I want to display the currently logged-in user in the navbar.

![image](https://github.com/vercel/next.js/assets/52977034/89041072-9196-4b3b-bca2-7ac5133d6182)

If I fetch the session server-side (`getServerSession` or now just `await auth()` in v5), **all pages** that display this navbar are **dynamically rendered**. There is no way to **statically cache** them anymore, correct?

Considering this, would it make sense to move the session retrieval **client-side**, if I want to enable static caching? This would be done by wrapping the layout into a `SessionProvider`, **not** passing a `session` prop, and then retrieving the session via `useSession` inside the navbar.

    <html lang="en">
      <body className={inter.className}>
        // Not passing the session prop here will make the provider fetch it     
    client-side
        <SessionProvider>
          <Navbar />
          {children}
        </SessionProvider>
      </body>
    </html>

Am I missing something? Like another better option? Or any caveats about my approach?

**Comment by devfraga** [comment]
Then it ends up becoming a mess, created some projects testing auth v5 and as soon as you use something inside the header that uses (await auth() from v5) all routes belonging to the layout become dynamic and the cache no longer works. Which basically takes away one of the fundamental features of nextjs such as (revalidate).

Have you come up with any solutions for this with Auth v5?

**Reply by florianwalther-private** [reply]
Yes, you can either use **client-side session fetching** as I explained (via `useSession` and a `SessionProvider` with **no** `session` prop) or wait for Next.js **partial prerendering**.

**Reply by devfraga** [reply]
Thx, The problem when I am using SessionProvider and useSession by the client after logout in, it is not reflected in my component unless I force an F5

Did you go through this?

**Reply by florianwalther-private** [reply]
The object returned from `useSession` has an `update` method. Call it without an argument to refetch the session client-side any time.
